mlrun/artifacts/base.py

```diff
@@ -967,6 +968,11 @@ def spec(self, spec):
         self._spec = self._verify_dict(spec, "spec", LinkArtifactSpec)
 
 
+# TODO: remove in 1.5.0
+@deprecated(
+    version="1.3.0",
+    reason="'LegacyArtifact' will be removed in 1.5.0, use 'Artifact' instead",
+)
```

Comment:
There are multiple inherited classes - `LegacyModelArtifact`, `LegacyDatasetArtifact` etc. - they should all be deprecated (sorry, the Jira ticket wasn't very clear on that...)